### ISSUE TITLE
Trying to use togglePopup()

### ISSUE THREAD

**Issue by anubisatari** [arnaudleclerc/ng-azure-maps#52]
When I try to call marker.togglePopup(), I receive the following error:
The marker must be added to a map before calling togglePopup().

Here is my map setup:
	<azure-map [center]="[Longitude, Latitude]"
			   [pitch]="0"
			   [showBuildingModels]="true"
			   [zoom]="16">
		<map-html-marker (onMouseUp)="HoverAndLeave($event, marker)"
						 *ngFor="let marker of PeopleMarkers | async"
						 [draggable]="false" [position]="marker.getOptions().position"
						 [text]="marker.getOptions().text">
		</map-html-marker>
	</azure-map>

Here is the PeopleMarkers setup:
	PeopleMarkers: Observable<CustomHtmlMarker[]>;

Here is my CustomHtmlMarker:
class CustomHtmlMarker extends atlas.HtmlMarker {
	constructor(public customOptions: HtmlMarkerOptions) {
		super(customOptions);
	}

I just needed to add a custom field for my use.

Here is the creation of a marker:
			const marker: CustomHtmlMarker = new CustomHtmlMarker({
				text: `${newperson.pwauser.firstName.substring(0, 1)}${newperson.pwauser.lastName.substring(0, 1)}`,
				position: [newperson.longitude, newperson.latitude],
				connectionid: connectionid,
				popup: new atlas.Popup({
					content: `<div style="padding: 10px;">${newperson.pwauser.firstName} ${newperson.pwauser.lastName}</div>`,
				}),
			});

Any help is appreciated. I am banging my head against the wall.

**Comment by arnaudleclerc** [comment]
Hi @anubisatari . The `map-html-marker` library creates a new `atlas.HtmlMarker` based on the given inputs. Your custom marker is never added to the map, that's why you receive this error message when you try to open the popup.

If you need a custom marker, the `map-html-marker` directive will have to expose a new `marker` input which will have priority other the other inputs. 

I don't see any reason why this could not be done. I will take a look at it as soon as possible.

**Comment by arnaudleclerc** [comment]
I just published a v5.1.0-alpha.21 of the package. This introduces a `[marker]` input on the `map-html-marker` directive.

If this input is provided, other inputs will be ignored as the library assumes all the options are already set on the marker. Following example opens a popup on click : 

```
import { Component } from '@angular/core';
import { IMarkerEvent } from 'ng-azure-maps';
import * as atlas from 'azure-maps-control';
import { Observable } from 'rxjs/internal/Observable';
import { of } from 'rxjs/internal/observable/of';

@Component({
  selector: 'app-root',
  template: '<azure-map (onReady)="mapReady()">' +
    '<map-html-marker *ngFor="let peopleMarker of peopleMarkers | async" [marker]="peopleMarker" (onMouseUp)="openPopup($event)">' +
    '</map-html-marker>' +
    '</azure-map>',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {

  public markerPositions: [number, number][] = [];
  public peopleMarkers: Observable<atlas.HtmlMarker[]>;

  mapReady() {
    for (let i = 0; i < 10; i++) {
      this.markerPositions.push([i * 5, i * 5]);
    }

    this.peopleMarkers = of(this.markerPositions.map(position => {
      const marker = new CustomMarker({
        position,
        popup: new atlas.Popup({
          content: `<div style="padding: 10px;">Test Popup</div>`,
        }),
      });
      marker.customerProperty = "myCustomValue";
      return marker;
    }))
  }

  public openPopup(event: IMarkerEvent): void {
    console.log((<CustomMarker>event.marker).customerProperty)
    event.marker.togglePopup();
  }
}

class CustomMarker extends atlas.HtmlMarker {

  public customerProperty: string;

  constructor(options: atlas.HtmlMarkerOptions) {
    super(options);
  }
}
```

Could you please give this version a try and let me know if this works in your case ? If so, I will publish a stable version later.

**Comment by anubisatari** [comment]
This works beautifully once I pay strict attention to the HTML in your response. I wasn't paying attention to, we changed [position] to [marker]. Works like a charm. I just need to style my popup a little more.
Thanks so much for your help.
anubisatari
